File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = ''
    for i in s:
        if i == " ":
            answer += " "
        elif i.islower():
            o = ord(i) + n
            if o > 122:
                answer += chr(o-26)
            else:
                answer += chr(o)
        else:
            o = ord(i) + n
            if o > 90:
                answer += chr(o-26)
            else:
                answer += chr(o)
    return answer

File: 프로그래머스/1/12926. 시저 암호/시저 암호.kt
class Solution {
    fun solution(s: String, n: Int): String {
        var answer = ""
        s.forEach { i -> 
            // 소문자일 경우 
            if (i.isLowerCase()) {
                val next = i.toInt() + n
                if (next > 122) {
                    answer += (next%122 + 96).toChar()
                }else {
                    answer += next.toChar()
                }
            }
            // 대문자일 경우 
            else if (i.isUpperCase()) {
                val next = i.toInt() + n
                if (next > 90) {
                    answer += (next%90 + 64).toChar()
                }else {
                    answer += next.toChar()
                }
            } else {
                answer += ' '
            }
        }
        return answer
    }
}
